crates/extensions/collectd_ext/src/converter.rs

```diff
@@ -0,0 +1,65 @@
+use crate::batcher::MessageBatch;
+use crate::collectd::CollectdMessage;
+use async_trait::async_trait;
+use batcher::BatchDriverOutput;
+use tedge_actors::ChannelError;
+use tedge_actors::DynSender;
+use tedge_actors::Sender;
+use tedge_mqtt_ext::MqttMessage;
+use tedge_mqtt_ext::Topic;
+use tracing::error;
+
+pub struct BatchOutputSender {
```

Comment:
This one feels like this  is an overloaded `Sender` which could have been an actor on its own, connected to the `MqttActor`.

Reply:
This is true. I'm using here a [bump on the wire](https://en.wikipedia.org/wiki/Bump-in-the-wire) while one can also use an actor that do a simple message conversion on each message received.

I think that both options should be available at the level of the `tedge_actors` crate.

Reply:
Yes, I completely agree. This bump on wire mechanics should be available as a first class citizen in the actors crate so that we can avoid writing heavy actors for doing quick mapping from one type to another.

Reply:
This has been simplified by https://github.com/thin-edge/thin-edge.io/pull/1880